### ISSUE TITLE
about the rmse of test when using NFM and ml-100k

### ISSUE THREAD

**Issue by 9ayhub** [analytics-ufcg/Deep4Rec#23]
I am a college student who is learning recommendation system. Thanks for your models which makes it easier for me to learn. However, When I use NFM and ml-100k to train, though the rmse of the training set can achieve good results（such as 0.3）, the rmse of the test set is still 0.9. Could you tell me where the problem is？thank you very much！

**Comment by mari-linhares** [comment]
Hey @9ayhub,

I'm happy that this repository is helping you with your studies!

> When I use NFM and ml-100k to train, though the rmse of the training set can achieve good results（such as 0.3）, the rmse of the test set is still 0.9. Could you tell me where the problem is？thank you very much！

Well, I can think of two main possible reasons:

1. Most probably the model is overfitting the dataset, from my own experiments + plots of the original paper, I think this model is very much prone to overfit. You can try different hyperparameters (dropout, batch norm, l2 reg) and architectures to avoid overfitting.
If you find a good architecture for this dataset let us know :)!

2. There could be issues with the code. We're still developing / testing methods so there are possible bugs in the code. Although, I've tested NFM with frappe dataset and I got similar results to the ones presented in the original paper.

Let us know if you have any updates on this.

Thank you :blush:

**Comment by mari-linhares** [comment]
I'll close this for now since is not related to any issues in the code at this point

**Comment by mari-linhares** [comment]
Hey @9ayhub,

I was investigating this issue and I think the `l2_regularization` is not being applied during training... Working on it...
